controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -195,6 +199,16 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
 	toImageRepository := parsedImage.Repository
 	if clusterConfig.ImageRepository != "" {
 		toImageRepository = strings.TrimSuffix(clusterConfig.ImageRepository, "/")
+	} else if kubernetesVersion.GTE(semver.Version{Major: 1, Minor: 25, Patch: 0, Pre: []semver.PRVersion{{VersionStr: "alpha"}}}) {
```

Comment:
Note: this would currently cause a rollout even if the image tag does not change for coredns.

Should we prevent updating the repository/registry if the coredns image tag does not change?

Reply:
I'm ok in having a rollout when only the imageRepository changes, this is a valid use case